HIP/hip-1056.md

```diff
@@ -0,0 +1,1769 @@
+filter on only accounts they are interested in. Block items removed from a filtered stream are represented by one or
+more `FilteredItemHash`’s which provide the missing hashes required to validate the filtered stream with the same block
+proof as the full stream.
+
+#### BlockHeader
+
+### BlockHeader
+
+A `BlockHeader` is one type of `output` BlockItem and will always be the first element in a Block. It contains Block
+metadata which describes a block’s identifying information such as its number and HAPI specification version
+
+```protobuf
+message BlockHeader {
+    /**
+     * Version of the HAPI specification that was used to serialize the block.
+     */
+    proto.SemanticVersion hapi_proto_version = 1;
+
+    /**
+     * The software version used to execute transactions in this block.
+     */
+    proto.SemanticVersion software_version = 2;
+
+    /**
+     * The block number of this block.
+     * <p>
+     * This value MUST be exactly `1` more than the previous block.<br/>
+     * Client systems SHOULD optimistically reject any block with a gap or
+     * reverse in `number` sequence, and MAY assume the block stream has
+     * encountered data loss, data corruption, or unauthorized modification.
+     */
+    uint64 number = 3;
+
+    /**
+     * The block hash of the previous block.
+     * This is here so a client parsing the BlockHeader can optimistically reject
+     * a block if this hash does not match the root hash of the previous block.
+     */
+    bytes previous_block_hash = 4;
+
+    /**
+     * A consensus timestamp for the start of this block.
+     * <p>
+     * This SHALL be the timestamp assigned by the hashgraph consensus
+     * algorithm to the first transaction of this block.
+     */
+    proto.Timestamp first_transaction_consensus_time = 5;
+
+    /**
+     * The hash algorithm used in this block.
+     */
+    proto.BlockHashAlgorithm hash_algorithm = 6;
+
+    /**
+     * Version of the active address book of the network at the time relating to the network public key
+     */
+    proto.SemanticVersion address_book_version = 7;
+}
+
+// This definition is in basic_types.proto.
+/**
+ * A specific hash algorithm used within a block.
+ *
+ * We did not reuse HashAlgorithm here because in all cases for now it will be SHA2_384 and if that's the default value
+ * then we can save space by not serializing it, whereas HASH_ALGORITHM_UNKNOWN = 0; is the default for HashAlgorithm.
+ */
+enum BlockHashAlgorithm {
+    SHA2_384 = 0;
+}
+```
+
+<aside>
+🚨 **Open Detail:** How do we define the `first_transaction_consensus_time` of a block with no transactions? Can it be
+first event consensus time if no transactions? what about empty block with no events?
+</aside>
+
+#### EventHeader
+
+The event header depicts the beginning of an event. All items after this and before the next `EventHeader` or
+`BlockProof` are part of this event or results of execution of this event’s contents. It is designed so that it is easy
```

Comment:
Wouldn't a third "event end" marker would be the non-transactional state changes?

Reply:
There is no mechanism for a reader to differentiate non-transactional state changes from transactional state changes.  To a reader of the Block Stream, those are just additional state changes in the event.

Is it important to differentiate those in some way?  We might be missing another header if that's the case...

Reply:
Ah, yes, that makes sense. I suppose the only part that is not correct is that if there are non-transaction state changes (i.e. like platform state changes or the sum of all changes to the transaction receipt queue), those are not a result of the last event's execution or contents. They are a consequence of the round as a whole. But I may be nit-picking wording here where not necessary. Feel free to close this comment if you think the distinction is not important.

Reply:
Good detail. I added a snippet based on your suggestion. Thanks